sources/packages/backend/libs/integrations/src/services/msfaa-number/msfaa-number.service.ts

```diff
@@ -196,17 +201,58 @@ export class MSFAANumberService extends RecordDataModelService<MSFAANumber> {
         "Not all required fields to update a received MSFAA record were provided.",
       );
     }
-
-    return this.repo.update(
-      {
-        msfaaNumber: msfaaNumber,
-        cancelledDate: IsNull(),
-        newIssuingProvince: IsNull(),
-      },
-      {
-        cancelledDate: getISODateOnlyString(cancelledDate),
-        newIssuingProvince,
-      },
-    );
+    return this.dataSource.transaction(async (transactionEntityManager) => {
+      // Keeping both MSFAA Cancellation update and saving of MSFAA Notification message as a part of the same transaction.
+      // Update the MSFAA record that needs to be cancelled.
+      const updateResult = await this.repo.update(
+        {
+          msfaaNumber: msfaaNumber,
+          cancelledDate: IsNull(),
+          newIssuingProvince: IsNull(),
+        },
+        {
+          cancelledDate: getISODateOnlyString(cancelledDate),
+          newIssuingProvince,
+        },
+      );
+      // In case no MSFAA record was cancelled.
+      if (!updateResult.affected) {
+        throw new Error(
+          `Error while cancelling MSFAA number: ${msfaaNumber}. Number of affected rows was ${updateResult.affected}, expected 1.`,
+        );
+      }
+      // Only if the msfaa record was successfully cancelled, then save the msfaa cancellation notification.
+      // This also takes care of the scenario where the msfaa record was already cancelled before in which case the record will not be updated and hence no notification will be saved.
+      if (updateResult.affected) {
```

Comment:
This condition is not required anymore. If code is past line 224 the updateResult.affected cannot be 0.

Reply:
Removed it. Thanks.